src/components/crud-operations.js

```diff
@@ -0,0 +1,214 @@
+import * as dom from './dom-elements';
+import LocalStorage from './local-storage';
+import Task from './task';
+import createTaskElement, {
+  allocateSpaceForToDOList,
+  getCheckedTaskElementId,
+} from './task-element-utils';
+
+export default class CRUD {
+  constructor() {
+    this.storageManagement = new LocalStorage();
+    this.listElement = dom.toDoList;
+    this.clearButton = dom.clearButton;
+    this.newTaskInput = dom.newTaskInput;
+    this.addButton = dom.addButton;
+    this.checkBoxes = document.querySelectorAll(
+      'li input[type="checkbox"]',
+    );
+    this.descriptionInputs = document.querySelectorAll(
+      'li input[type="text"]',
+    );
+    this.deleteButtons = document.querySelectorAll(
+      'li button.delete-button',
+    );
+    this.moveButtons = document.querySelectorAll(
+      'li button.move-button',
+    );
+
+    // event listeners
+    this.addButton.addEventListener(
+      'click',
+      this.onAddButtonClicked,
+    );
+
+    this.clearButton.addEventListener(
+      'click',
+      this.onClearButtonClicked,
+    );
+  }
+
+  doOnCheckboxChecked = (e) => {
+    const id = e.target.id.split('-')[1];
+    const task = this.storageManagement.readLocalStorage()[id];
+    this.storageManagement.changeTaskStatus(
+      task,
+      e.target.checked,
+    );
+  };
+
+  doOnDescriptionInputChanged = (e) => {
+    const id = e.target.id.split('-')[1];
+    const task = this.storageManagement.readLocalStorage()[id];
+    this.storageManagement.changeTaskDescription(
+      task,
+      e.target.value,
+    );
+  };
+
+  doOnDeleteButtonClicked = (e) => {
+    const button = e.currentTarget;
+    if (button.id.split('-')[0] === 'deleteButton') {
+      const id = button.id.split('-')[1];
+      const remainingTasks = this.storageManagement
+        .readLocalStorage()
+        .filter((task) => task.index !== +id);
+      this.storageManagement.updateLocalStorage(
+        remainingTasks,
+      );
+      this.displayUpdatedList();
+    }
+  };
```

Comment:
- [ ] You might need to check up on this function. When a task is deleted, the indexes of the remaining ones are re-order. From the screenshots below, I added four tasks, deleted the third one. After the delete operation, the fourth task with index 3 should now be index 2. But as of now, it doesn't work like that. Please check it out.


Before Delete |
:-----------------:|
![p1](https://user-images.githubusercontent.com/39523312/154367172-960a555e-6e80-4642-be6f-99a23dc3eab8.png)

 After Delete   |
:-----------------:|
![p2](https://user-images.githubusercontent.com/39523312/154367181-f092b032-fe50-410b-85a9-35d6a96abd60.png)

Reply:
Hi @mmsesay ! :wave: 
Can you please try again? Because It's working at my end on both locally or using the GitHub pages link.
Perhaps, you tried the GitHub pages yesterday and the changes did not take effect yet!
Looking forward to your replay
Thank you

Reply:
OK, nevermind
I understand what you mean now
Thanks for pointing that out
:+1: